src/fastoad/openmdao/problem.py

```diff
@@ -199,12 +199,14 @@ def _get_problem_inputs(self) -> Tuple[VariableList, VariableList]:
         :return: VariableList of needed input variables, VariableList with unused variables.
         """
 
-        problem_inputs_names = [var.name for var in self.analysis.problem_variables if var.is_input]
-
         input_file_variables = DataFile(self.input_file_path)
 
         unused_variables = VariableList(
-            [var for var in input_file_variables if var.name not in problem_inputs_names]
+            [
+                var
+                for var in input_file_variables
+                if var.name not in self.analysis.problem_variables.names()
+            ]
```

Comment:
If I understand correctly, this is the heart of this PR ? The ouputs of the problem are not considered unused anymore so they will be set in the `_set_input_values_post_setup` or `_set_input_values_pre_setup` methods ? Which raises the question of whether those name should be kept because output values can also be set now. Or maybe those method should be rename to `_set_input_file_values...`

Reply:
Well, we still talk about input **values**. It's just that they now concern also output **variables** 😉